dask/dataframe/tests/test_shuffle.py

```diff
@@ -916,3 +916,47 @@ def test_set_index_timestamp():
 
     assert_eq(df2, ddf_new_div)
     assert_eq(df2, ddf.set_index("A"))
+
+
+@pytest.mark.parametrize("compression", [None, "ZLib"])
+def test_disk_shuffle_with_compression_option(compression):
+    # test if dataframe shuffle works both with and without compression
+    with dask.config.set({"dataframe.shuffle-compression": compression}):
+        test_shuffle("disk")
+
+
+@pytest.mark.parametrize("compression", ["UNKOWN_COMPRESSION_ALGO"])
+def test_disk_shuffle_with_unknown_compression(compression):
+    # test if dask raises an error in case of fault config string
+    with dask.config.set({"dataframe.shuffle-compression": compression}):
+        with pytest.raises(ImportError) as excinfo:
+            test_shuffle("disk")
+        assert str(excinfo.value) == (
+            "Not able to import and load {0} as compression algorithm."
+            "Please check if the library is installed and supported by Partd.".format(
+                compression
+            )
+        )
+
+
+def test_disk_shuffle_check_actual_compression():
+    # test if the compression switch is really respected by testing the size of the actual partd-data on disk
+    def generate_raw_partd_file(compression):
+        # generate and write a dummy dataframe to disk and return the raw data bytes
+        df1 = pd.DataFrame({"a": list(range(10000))})
+        df1["b"] = (df1["a"] * 123).astype(str)
+        with dask.config.set({"dataframe.shuffle-compression": compression}):
+            f = maybe_buffered_partd(buffer=False, tempdir=None)
+            p1 = f()
+            p1.append({"x": df1})
+            # get underlying filename from partd - depending on nested structure of partd object
+            filename = (
+                p1.partd.partd.filename("x") if compression else p1.partd.filename("x")
+            )
+            return open(filename, "rb").read()
+
+    # get compressed and uncompressed raw data
+    uncompressed_data = generate_raw_partd_file(compression=None)
+    compressed_data = generate_raw_partd_file(compression="BZ2")
+
+    assert len(uncompressed_data) > len(compressed_data)
```

Comment:
Perhaps rather than this assert, something like

```python
with dask.config.set({"dataframe.shuffle-compression": "BZ2"}):
    p = maybe_buffered_partd(...)()
    assert isinstance(p, partd.encode.Encode)
    
```

IIUC, that should more easily distinguish whether the config value is taking effect. Then we don't have to worry about encoding possibly inflating the file size.

Reply:
a) In this case we do not test anymore if compression is actually applied, as you proposed earlier.
b) Both (compression=None) and (compression="BZ2") returns a partd.encode.Encode object so we can not distinguish between the cases
c) Any other idea? Currently this is a more or less a workaround because we have no interface to partd which exposes the (compression) properties of encode object. Or is there any? This would be, I guess, the most clean way.

Reply:
I would have thought you got a `partd.File` with compression=None. Is that not the case?

Reply:
Unfortunatly not.
`maybe_buffered_partd` returns `partd.PandasBlocks`, which is always type `partd.encode.Encode`

```python
with dask.config.set({"dataframe.shuffle-compression": None}):
    f = maybe_buffered_partd(buffer=False, tempdir=None)
    print(type(f))
    print(type(f()))

<class 'dask.dataframe.shuffle.maybe_buffered_partd'>
<class 'partd.encode.Encode'>
```

and
```python
with dask.config.set({"dataframe.shuffle-compression": "BZ2"}):
    f = maybe_buffered_partd(buffer=False, tempdir=None)
    print(type(f))
    print(type(f()))

<class 'dask.dataframe.shuffle.maybe_buffered_partd'>
<class 'partd.encode.Encode'>
```

Also, `maybe_buffered_partd` does not expose the original file object.

Reply:
Gotcha, thanks.

Reply:
Welcome :)